ci/include_grouping.py

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env python3
+"""
+easy command line to run against all citus-style checked files:
+
+$ git ls-files \
+  | git check-attr --stdin citus-style \
+  | grep 'citus-style: set' \
+  | awk '{print $1}' \
+  | cut -d':' -f1 \
+  | xargs -n1 ./ci/include-grouping.py
+"""
+
+import collections
+import os
+import sys
+
+
+def main(args):
+    if len(args) < 2:
+        print("Usage: include-grouping.py <file>")
+        return
+
+    file = args[1]
+    if not os.path.isfile(file):
+        print("File does not exist", file=sys.stderr)
+        return sys.exit(1)
+
+    with open(file, "r") as f:
+        with open(file + ".tmp", "w") as out_file:
+            lines = f.readlines()
+            includes = []
+            skipped_lines = []
+
+            # This calls print_sorted_includes on a set of consecutive #include lines.
+            # This implicitly keeps separation of any #include lines that are contained in
+            # an #ifdef, because it will order the #include lines inside and after the
+            # #ifdef completely separately.
+            for line in lines:
+                # if a line starts with #include we don't want to print it yet, instead we
+                # want to collect all consecutive #include lines
+                if line.startswith("#include"):
+                    includes.append(line)
+                    skipped_lines = []
+                    continue
+
+                # if we have collected any #include lines, we want to print them sorted
+                # before printing the current line. However, if the current line is empty
+                # we want to perform a lookahead to see if the next line is an #include.
+                # To maintain any separation between #include lines and their subsequent
+                # lines we keep track of all lines we have skipped inbetween.
+                if len(includes) > 0:
+                    if len(line.strip()) == 0:
+                        skipped_lines.append(line)
+                        continue
+
+                    # we have includes that need to be grouped before printing the current
+                    # line.
+                    print_sorted_includes(includes, file=out_file)
+                    includes = []
+
+                # print any skipped lines
+                print("".join(skipped_lines), end="", file=out_file)
+                skipped_lines = []
```

Comment:
I think this block logically should be inside `if len(includes) > 0:`, because we need to collect and print skipped lines only when there are some collected includes.
```suggestion
                    # print any skipped lines
                    print("".join(skipped_lines), end="", file=out_file)
                    skipped_lines = []
```